sdk/src/client/api/block_builder/input_selection/requirement/mana.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{Error, InputSelection};
+use crate::client::secret::types::InputSigningData;
+
+impl InputSelection {
+    pub(crate) fn fulfill_mana_requirement(&mut self, allotments: u64) -> Result<Vec<InputSigningData>, Error> {
+        let required_mana = self.outputs.iter().map(|o| o.mana()).sum::<u64>() + allotments;
+        let mut selected_mana = self.selected_inputs.iter().map(|o| o.output.mana()).sum::<u64>();
```

Comment:
Here we need to add the mana over time - decay?

Reply:
Do you allow me to do that in another PR? I'd apply potential/decay wherever relevant in a single PR

Reply:
Sure, actually prefer that, as long as we don't forget about it